pkg/bootflow/examples/ocpdxepcr0/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/9elements/converged-security-suite/v2/pkg/bootflow/bootengine"
+	"github.com/9elements/converged-security-suite/v2/pkg/bootflow/flows"
+	"github.com/9elements/converged-security-suite/v2/pkg/bootflow/systemartifacts/biosfirmware"
+	"github.com/9elements/converged-security-suite/v2/pkg/bootflow/trustchains/tpm"
+	"github.com/9elements/converged-security-suite/v2/pkg/bootflow/types"
+	"github.com/google/go-tpm/tpm2"
+)
+
+func main() {
+	// parsing arguments
+	flag.Parse()
+	biosFirmwarePath := flag.Arg(0)
+	biosFirmware, err := os.ReadFile(biosFirmwarePath)
+	if err != nil {
+		panic(fmt.Errorf("unable to read BIOS firmware image '%s': %w", biosFirmwarePath, err))
+	}
+
+	// the main part
+	state := types.NewState()
+	state.SetFlow(flows.OCPDXE(), 0)
+	state.EnableTrustChain(tpm.NewTPM())
+	state.EnableSystemArtifact(biosfirmware.NewBIOSFirmware(biosFirmware))
+	process := bootengine.NewBootProcess(state)
+	process.Finish()
+
+	// printing results
+	tpm.StateExec(state, func(t *tpm.TPM) error {
```

Comment:
design nit; are there any alternatives to using this state object as an opaque pointer? this reads a bit weird to me, although it does address extensibility (which i assume was the main reason for this form in the first place)

Reply:
Could you rephrase, please?

Do I understand correctly the question is: can we avoid this `StateExec` function? If so, the answer is yes:
If above we would do:
```
tpmInstance := tpm.NewTPM()
state.EnableTrustChain(tpmInstance)
```
instead of:
```
state.EnableTrustChain(tpm.NewTPM())
```
then we could've just access `tpmInstance` directly without calling `StateExec`.

Reply:
oh... so that's the semantic. Wasnt clear to me from the naming "StateExec". Well, why not have a "GetChain" instead of the lambda call? What's the benefit here? (i think you have a similar pattern for Log)

Reply:
It was just for syntax sugar.

Talked with @mimir-d in a video chat. Decided to go with `GetChain` instead.

Reply:
Changed my mind. Since now we migrated CSS to go1.18, now we can use generics, so in the end I'm going with:
```
tpmInstance := tpm.GetFrom(state)
```
instead of:
```
tpm.StateExec(state, func(tpmInstance *tpm.TPM) error {
```